xonsh/xonfig.py

```diff
@@ -110,6 +112,76 @@
 Thanks for using the xonsh configuration wizard!"""
 
 
+@lazyobject
+def _XONFIG_SOURCE_FOREIGN_SHELL_COMMAND():
+    return collections.defaultdict(lambda: 'source-foreign',
```

Comment:
I'm pretty sure that `defaultdict` is as cheap or cheaper than `lazyobject`. This isn't doing IO or complex calculation.

Reply:
```
🐚 timeit("lazyobject(lambda: None)", "from xonsh.lazyasd import lazyobject")
0.5423036129999446
🐚 timeit("defaultdict(lambda: 'source-foreign', bash='source-bash', cmd='source-cmd', zsh='source-zsh')", "from collections import defaultdict")
0.385492806000002
```

Reply:
oh yeah, `defaultdict` is in C.

https://github.com/python/cpython/blob/c5ae169e1b73315672770517bf51cf8464286c76/Lib/collections/__init__.py#L42

Reply:
You are right!

Reply:
Changed.